### PR TITLE
Allow skipping null byte getting appended in syslog message

### PR DESCRIPTION
**Pull request by ruhul-sophos**

### CODE REVIEW

**Review by rkamat — APPROVED** [review]

File: api_client.py [hunk 1]
```diff
@@ -160,6 +160,7 @@ def add_siem_logeer_handler(self, logdir):
             logging_handler = logging.FileHandler(
                 os.path.join(logdir, self.config.filename), "a", encoding="utf-8"
             )
+        logging_handler.append_nul = self.config.append_nul == "true"
```
[the remark this answers is not in the document]
Comment (ruhul-sophos) [reply api_client.py]: done

**Comment by schuberttobias** [comment]
@rkamat: This PR fixes a problem we're currently experiencing: Elastic Kibana adheres to RFC3164 and expects no nul byte in syslog messages. Currently we're using Sophos-Central-SIEM-Integration v2.1.0 and the syslog messages include null bytes which leads to Kibana failing to parse them correctly.

Could you please release a new version (e.g. v2.1.1) including this PR?

**Comment by schuberttobias** [comment]
@rkamat: Please ignore my last message, sorry for the noise. The nul byte seems to be unrelated. I am currently investigating what causes the issue in Kibana .

**Comment by rkamat** [comment]
@schuberttobias No worries, thanks for the update.

**Pull request merged**